Windows tests fail on installing msprime

See for instance https://github.com/MesserLab/SLiM/runs/5028497404?check_suite_focus=true where we get:
```
Run C:\Miniconda\condabin\conda.bat activate anaconda-client-env
  C:\Miniconda\condabin\conda.bat activate anaconda-client-env
  git clone https://github.com/tskit-dev/pyslim.git
  cd pyslim && pip install .
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
  env:
    MSYSTEM: MINGW64
Cloning into 'pyslim'...
Processing d:\a\slim\slim\pyslim
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting msprime>=1.0.1
  Downloading msprime-1.1.0.tar.gz (850 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Installing backend dependencies: started
  Installing backend dependencies: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting tskit
  Downloading tskit-0.4.1-cp37-cp37m-win_amd64.whl (394 kB)
Collecting kastore
  Downloading kastore-0.3.1-cp37-cp37m-win_amd64.whl (29 kB)
Collecting numpy
  Downloading numpy-1.21.5-cp37-cp37m-win_amd64.whl (14.0 MB)
Collecting newick>=1.3.0
  Downloading newick-1.3.2-py2.py3-none-any.whl (12 kB)
Collecting demes>=0.2
  Downloading demes-0.2.1-py3-none-any.whl (38 kB)
Collecting svgwrite>=1.1.10
  Downloading svgwrite-1.4.1-py3-none-any.whl (66 kB)
Collecting jsonschema>=3.0.0
  Downloading jsonschema-4.4.0-py3-none-any.whl (72 kB)
Collecting humanize
  Downloading humanize-3.14.0-py3-none-any.whl (98 kB)
Collecting attrs>=20.3.0
  Downloading attrs-21.4.0-py2.py3-none-any.whl (60 kB)
Collecting ruamel.yaml>=0.15.78
  Downloading ruamel.yaml-0.17.20-py3-none-any.whl (109 kB)
Collecting importlib-resources>=1.4.0
  Downloading importlib_resources-5.4.0-py3-none-any.whl (28 kB)
Collecting pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0
  Downloading pyrsistent-0.18.1-cp37-cp37m-win_amd64.whl (61 kB)
Requirement already satisfied: importlib-metadata in c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages (from jsonschema>=3.0.0->tskit->pyslim==0.700) (4.10.1)
Requirement already satisfied: typing-extensions in c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages (from jsonschema>=3.0.0->tskit->pyslim==0.700) (4.0.1)
Requirement already satisfied: zipp>=3.1.0 in c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages (from importlib-resources>=1.4.0->jsonschema>=3.0.0->tskit->pyslim==0.700) (3.7.0)
Collecting ruamel.yaml.clib>=0.2.6
  Downloading ruamel.yaml.clib-0.2.6-cp37-cp37m-win_amd64.whl (115 kB)
Using legacy 'setup.py install' for pyslim, since package 'wheel' is not installed.
Building wheels for collected packages: msprime
  Building wheel for msprime (pyproject.toml): started
  Building wheel for msprime (pyproject.toml): finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: 'c:\hostedtoolcache\windows\python\3.7.9\x64\python.exe' 'c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py' build_wheel 'C:\Users\RUNNER~1\AppData\Local\Temp\tmpy49d6xhp'
       cwd: C:\Users\RUNNER~1\AppData\Local\Temp\pip-install-dmlt1kdr\msprime_c4f4d41c514c4ae0aa44ad7f7542fe94
  Complete output (41 lines):
  running bdist_wheel
  running build
  running build_py
  running egg_info
  writing msprime.egg-info\PKG-INFO
  writing dependency_links to msprime.egg-info\dependency_links.txt
  writing entry points to msprime.egg-info\entry_points.txt
  writing requirements to msprime.egg-info\requires.txt
  writing top-level names to msprime.egg-info\top_level.txt
  setup.py:27: UserWarning: Error occured getting GSL path config: [WinError 2] The system cannot find the file specified
    warn(f"Error occured getting GSL path config: {e}")
  listing git files failed - pretending there aren't any
  reading manifest file 'msprime.egg-info\SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  adding license file 'LICENSE'
  running build_ext
  building 'msprime._msprime' extension
  cl : Command line warning D9002 : ignoring unknown option '-std=c99'
  kastore.c
  git-submodules\tskit\c\subprojects\kastore\kastore.c(28): warning C4996: 'strerror': This function or variable may be unsafe. Consider using strerror_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
  git-submodules\tskit\c\subprojects\kastore\kastore.c(568): warning C4996: 'fopen': This function or variable may be unsafe. Consider using fopen_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
Failed to build msprime
  cl : Command line warning D9002 : ignoring unknown option '-std=c99'
  core.c
  git-submodules\tskit\c\tskit\core.c(139): warning C4996: 'strerror': This function or variable may be unsafe. Consider using strerror_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
  cl : Command line warning D9002 : ignoring unknown option '-std=c99'
  tables.c
  git-submodules\tskit\c\tskit\tables.c(206): warning C4996: 'strcpy': This function or variable may be unsafe. Consider using strcpy_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
  git-submodules\tskit\c\tskit\tables.c(207): warning C4996: 'strcat': This function or variable may be unsafe. Consider using strcat_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
  git-submodules\tskit\c\tskit\tables.c(358): warning C4996: 'strcpy': This function or variable may be unsafe. Consider using strcpy_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
  git-submodules\tskit\c\tskit\tables.c(359): warning C4996: 'strcat': This function or variable may be unsafe. Consider using strcat_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
  git-submodules\tskit\c\tskit\tables.c(10670): warning C4996: 'fopen': This function or variable may be unsafe. Consider using fopen_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
  git-submodules\tskit\c\tskit\tables.c(10752): warning C4996: 'fopen': This function or variable may be unsafe. Consider using fopen_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
  cl : Command line warning D9002 : ignoring unknown option '-std=c99'
  trees.c
  cl : Command line warning D9002 : ignoring unknown option '-std=c99'
  avl.c
  lib\avl.c(223): warning C4098: 'avl_free_nodes': 'void' function returning a value
  cl : Command line warning D9002 : ignoring unknown option '-std=c99'
  fenwick.c
  lib\fenwick.c(31): fatal error C1083: Cannot open include file: 'gsl/gsl_math.h': No such file or directory
  error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
  ----------------------------------------
  ERROR: Failed building wheel for msprime
ERROR: Could not build wheels for msprime, which is required to install pyproject.toml-based projects
WARNING: You are using pip version 21.3.1; however, version 22.0.2 is available.
You should consider upgrading via the 'c:\hostedtoolcache\windows\python\3.7.9\x64\python.exe -m pip install --upgrade pip' command.
Error: Process completed with exit code 1.
```

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -227,8 +227,7 @@ jobs:
       with:
         path: |
           c:\Miniconda\envs\anaconda-client-env
-          /usr/share/miniconda/envs/anaconda-client-env
-          ~/osx-conda
+          ~/.bash_profile
           ~/.profile
         key: ${{ runner.os }}-${{ matrix.python}}-conda-v16-${{ hashFiles('treerec/tests/conda-requirements.txt') }}-${{ hashFiles('treerec/tests/pip-requirements.txt') }}
 
@@ -290,8 +289,8 @@ jobs:
     - name: Treesequence tests
       shell: bash -l {0}
       run: |
-        export PATH=$PATH:$(pwd)\Release
-        cd treerec\tests 
+        export PATH=$PATH:$(pwd)/Release
+        cd treerec/tests 
         python -m pytest -xv
 
   tests-windows-latest-GUI:

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -245,25 +245,20 @@ jobs:
 
     - name: Install conda deps
       if: steps.cache.outputs.cache-hit != 'true'
-      shell: pwsh
+      shell: bash -l {0}
       run: |
-        C:\Miniconda\condabin\conda.bat activate anaconda-client-env
-        C:\Miniconda\condabin\conda.bat install --yes --file=treerec/tests/conda-requirements.txt
-        C:\Miniconda\condabin\conda.bat init powershell
+        conda install --yes --file=treerec/tests/conda-requirements.txt
 
     - name: Install pip deps
       if: steps.cache.outputs.cache-hit != 'true'
-      shell: pwsh
+      shell: bash -l {0}
       run: |
-        C:\Miniconda\condabin\conda.bat activate anaconda-client-env
         pip install -r treerec/tests/pip-requirements.txt
 
     - name: Install pyslim
-      shell: pwsh
+      shell: bash -l {0}
       run: |
-        C:\Miniconda\condabin\conda.bat activate anaconda-client-env
-        C:\Miniconda\condabin\conda.bat init powershell
-        C:\Miniconda\condabin\conda.bat install -c conda-forge pyslim
+        conda install -c conda-forge pyslim
 
     - name: Debug
       run: |
@@ -293,10 +288,9 @@ jobs:
         ./slim -testSLiM
     
     - name: Treesequence tests
-      shell: pwsh
+      shell: bash -l {0}
       run: |
-        C:\Miniconda\condabin\conda.bat activate anaconda-client-env
-        $env:path += ";$(pwd)\Release"
+        export PATH=$PATH:$(pwd)\Release
         cd treerec\tests 
         python -m pytest -xv
 

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -327,6 +327,7 @@ jobs:
 
     - name: Build (cmake)
       run: |
+        cd .
         cd windows_compat/gnulib
         touch --date="`date`" aclocal.m4 Makefile.am configure configure.ac config.h.in Makefile.in
         cd ../..

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -230,7 +230,7 @@ jobs:
           /usr/share/miniconda/envs/anaconda-client-env
           ~/osx-conda
           ~/.profile
-        key: ${{ runner.os }}-${{ matrix.python}}-conda-v15-${{ hashFiles('treerec/tests/conda-requirements.txt') }}-${{ hashFiles('treerec/tests/pip-requirements.txt') }}
+        key: ${{ runner.os }}-${{ matrix.python}}-conda-v16-${{ hashFiles('treerec/tests/conda-requirements.txt') }}-${{ hashFiles('treerec/tests/pip-requirements.txt') }}
 
     - name: Install Conda
       uses: conda-incubator/setup-miniconda@v2

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -247,6 +247,7 @@ jobs:
       if: steps.cache.outputs.cache-hit != 'true'
       shell: pwsh
       run: |
+        C:\Miniconda\condabin\conda.bat activate anaconda-client-env
         C:\Miniconda\condabin\conda.bat install --yes --file=treerec/tests/conda-requirements.txt
         C:\Miniconda\condabin\conda.bat init powershell
 
@@ -261,6 +262,7 @@ jobs:
       shell: pwsh
       run: |
         C:\Miniconda\condabin\conda.bat activate anaconda-client-env
+        C:\Miniconda\condabin\conda.bat init powershell
         C:\Miniconda\condabin\conda.bat install -c conda-forge pyslim
 
     - name: Debug

